### DISCUSSION TITLE
[Help Request] How to get REGEX working properly?

### DISCUSSION THREAD

**Discussion by Xelbayria** [ScoopInstaller/Main#3823] [General]
This is from the website [Sumatrapdf reader Prerelease](https://www.sumatrapdfreader.org/prerelease) the HTML code below 
```HTML
<span id="buildver">15229</span>
```
I used "Inspector" to look at the HTML code
The REGEX is ` "regex": "<span id=\"\"buildver\"\">([\\d]+)</span>" `

I ran `~/bin/checkver.ps1` to test it and couldn't get it working because of this error: 

```couldn't match '<span id=""buildver"">([\d]+)</span>' in https://www.sumatrapdfreader.org/prerelease```

Changed the REGEX to ` "regex": "<span id=\"buildver\">([\\d]+)</span>" ` 
Above also got the error:

```couldn't match '<span id="buildver">([\d]+)</span>' in https://www.sumatrapdfreader.org/prerelease```

I have created different types of REGEXs. I am sure the REGEXs are created correctly and should be working properly. I had no problem with other websites where I want to update software in my own bucket. This one makes no sense as I couldn't figure out why. I appreciate your help with this.

**Comment by rashil2000** [comment]
The span you're trying to access is populated by JS when the page loads, so it's not present in the HTML.

Use this instead: https://kjkpubsf.sfo2.digitaloceanspaces.com/software/sumatrapdf/sumatralatest.js

**Reply by Xelbayria** [reply]
> The span you're trying to access is populated by JS when the page loads, so it's not present in the HTML.
> 
> Use this instead: https://kjkpubsf.sfo2.digitaloceanspaces.com/software/sumatrapdf/sumatralatest.js

eh? that was a JS? 👀 and thank you. I'll take a look at the link
🤔 how did you get the link from? this is something I would like to learn how to do on my own.

**Reply by rashil2000** [reply]
When visiting a web page, you can right-click and go to 'View Page Source', to see what the Regex will see. 

In this case the buildver was not present in that source, but there was a link to a JS file at the beginning, which contained the buildver.

**Comment by Xelbayria** [comment]
> When visiting a web page, you can right-click and go to 'View Page Source', to see what the Regex will see.
In this case the buildver was not present in that source, but there was a link to a JS file at the beginning, which contained the buildver.

dang. I checked "View Page Source" and it seriously doesn't have any numbers. 👀 Now I could see why the REGEXs I created are not working.  * scratching my head * 
Is there another way? 

-------------------
<strike> A quick question, how do I get the `scoop status` to stop showing the "Manifest removed" message? Because I removed a bucket and add a JSON file into my own bucket. It's alright if you don't know the answer to that question. </strike>

Nvm. I tried an inspiration that just came to me and that was the answer to my question.

**Reply by niheaven** [reply]
F12 - Developer Mode.

Check any files that downloaded locally and find the version number (maybe in JSON format). But this is a bit complex...

**Comment by rashil2000** [comment]
> Is there another way?

I don't think so. You will have to use another website.

**Reply by Xelbayria** [reply]
> I don't think so. You will have to use another website.

Alright, Thank you. I understand now why it doesn't work. It's disappointing and have a good day!.
